lib/api.rb

```diff
@@ -0,0 +1,7 @@
+module Api
+  class AuthenticationError < StandardError; end
```

Comment:
Another common pattern that I find very useful is the following:

```ruby
class ApiError < StandardError; end

class AuthenticationError < ApiError; end
class Forbidden < ApiError; end
...
````

This way all API errors are rescuable with `rescue ApiError`